pkg/apis/awsprovider/v1alpha1/awsclusterproviderconfig_types.go

```diff
@@ -122,18 +122,18 @@ type SubnetSpec struct {
 	// AvailabilityZone defines the availability zone to use for this subnet in the cluster's region.
 	AvailabilityZone string `json:"availabilityZone,omitempty"`
 
-	// IsPublic defines the subnet as a public subnet. Refer to the AWS documentation for further information.
+	// IsPublic defines the subnet as a public subnet. A subnet is considered public when it has a direct route to and from the internet via an internet gateway.
```

Comment:
This should be reworded to just `direct route to the internet via an internet gateway`. I don't think there is a way to check the route _from_ the igw to the VPC.

Reply:
I intended to differentiate an "igw" route from an "Egress-only Internet Gateway," with my assumption being that the purpose of the former is to somehow teach the AWS SDN a route from the internet into the subnet.

Maybe `when it has a route associated with a bidirectional internet gateway`?

Reply:
Is there any way to actually check that though?

Reply:
I believe the purpose of an internet gateway is to be bidirectional, but maybe the thing to do is take the phrasing straight from the [AWS docs](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Internet_Gateway.html):

> If your subnet is associated with a route table that has a route to an internet gateway, it's known as a public subnet.